### ISSUE TITLE
Issue accessing box_loss, cls_loss, and dfl_loss in on_train_epoch_end callback

### ISSUE THREAD

**Issue by pathu999** [ultralytics/ultralytics#13407]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Description:
I'm trying to access box_loss, cls_loss, and dfl_loss during the training of a YOLOv8 model using a callback function. However, the values printed are not as expected. Below is a simplified version of the code I'm using:

code:-
from ultralytics import YOLO

def on_train_epoch_end(trainer):
    metrics = trainer.metrics
    keys,vals = list(metrics.keys()), list(metrics.values())
    print("Values:", vals)
    
model = YOLO("yolov8n.yaml") 

model.add_callback("on_train_epoch_end", on_train_epoch_end)
results = model.train(data="data.yaml", epochs=3, imgsz=640)

yolov5                        7.0.13
yolov9                        0.0.1
Python version: 3.8.1
Operating System: windows

**Comment by glenn-jocher** [comment]
Hello,

Thank you for reaching out and providing detailed information about your issue. It seems like you're trying to access specific loss metrics within the `on_train_epoch_end` callback, but the values aren't as expected.

One potential issue could be how the metrics are being accessed or stored during training. Ensure that the loss values (`box_loss`, `cls_loss`, `dfl_loss`) are being correctly updated and logged within your training loop. These values should be part of the `trainer.metrics` dictionary if they are properly computed and stored during each training step.

If you're still facing issues, it might be helpful to verify the following:
1. Ensure that the loss calculations are being logged correctly in your training loop.
2. Check if any modifications in the model configuration or training script could be affecting how metrics are logged.
3. Consider adding more detailed logging within your callback function to trace how values change over epochs.

If the problem persists, please provide more details about the modifications you've made to the training loop or any additional configurations. This will help in diagnosing the issue more effectively.
